src/dst/bind.cpp

```diff
@@ -522,8 +524,8 @@ static std::unique_ptr<Expr> expand_patterns(const std::string &fnname,
             "_ a" + std::to_string(--var), DefValue(FRAGMENT_CPP_LINE, std::move(gets[i - 1]))));
         assert(out.second);
       }
-      Lambda *lam = new Lambda(rmap->body->fragment, "_ tuple_case", rmap.get());
-      rmap.release();
+      DefMap *unowned_rmap = rmap.release();
+      Lambda *lam = new Lambda(unowned_rmap->body->fragment, "_ tuple_case", unowned_rmap);
```

Comment:
The only claim of these being better is that clangd doesn't complain.

Before we had the warning

`value of release() should not be discarded`

which is true because that would be a memory leak (though we sneakily avoid it by capturing the pointer with `get()`. 

This update fixes the warning and makes it more explicit of what is happening/intended.

Reply:
A few weeks ago there was a segfault on fedora 36 because of the line

```
Lambda *lambda = new Lambda(expr->fragment, ast.name, expr.release());
```

which was due to UB on the order that application parameters are resolved. (crash happens when `expr.release()` happens before `expr->fragment`)

to avoid this is was re-written as 

```
Lambda *lambda = new Lambda(expr->fragment, ast.name, expr.get());
expr.release();
```

but that standalone `expr.release()` looks _a lot_ like a memory leak so a warning is triggered.

Reply:
I think assigning the fragment first is better. That's what you would do if we were using unique_ptr correctly

```
auto frag = expr->fragment;
Lambda *lambda = new Lambda(frag, ast.name, expr.release()); // not this would be std::move(expr) if we did things correctly
```

Reply:
I think that makes since for some of them. There are a couple ones where we create the unique_ptr just to immediatly throw it away though.

Legit something like

```
{
  std::unique_prt<Expr> expr = std::unique_ptr(new Expr(...));
  Lambda *l = new Lambda(frag, ast.name, expr.release());
}
```
in those cases we should just create the pointer until Lambda accepts a unique_ptr instead of a raw ptr

Reply:
Yeah ideally all the constructors would just take unique_ptr and we would just move the unique ptrs...bamn no memory leaks!

Reply:
Don't get me wrong, I 100% agree. I just don't want to refactor that right now. You are welcome to however :)